jme3-core/src/com/jme3/gde/core/util/datatransfer/AnimationDataFromOriginal.java

```diff
@@ -0,0 +1,121 @@
+package com.jme3.gde.core.util.datatransfer;
+
+import com.jme3.anim.AnimClip;
+import com.jme3.anim.AnimComposer;
+import com.jme3.anim.Armature;
+import com.jme3.anim.SkinningControl;
+import com.jme3.gde.core.scene.ApplicationLogHandler;
+import com.jme3.gde.core.util.TaggedSpatialFinder;
+import com.jme3.scene.SceneGraphVisitor;
+import com.jme3.scene.Spatial;
+
+import java.util.Collection;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Copies AnimComposer and AnimClips from an updated spatial to the original.
+ */
+public final class AnimationDataFromOriginal implements SpatialDataTransferInterface {
+
+    private static final Logger LOGGER
+            = Logger.getLogger(AnimationDataFromOriginal.class.getName());
+
+    private final TaggedSpatialFinder finder;
+
+    public AnimationDataFromOriginal(final TaggedSpatialFinder finder) {
+        this.finder = finder;
+    }
+
+    @Override
+    public void update(final Spatial root, final Spatial original) {
+        //loop through original to also find new AnimControls, we expect all 
+        // nodes etc. to exist
+        removeAnimData(root);
+        original.depthFirstTraversal(new SceneGraphVisitor() {
+
+            @Override
+            public void visit(final Spatial spatial) {
+                final AnimComposer animComposer
+                        = spatial.getControl(AnimComposer.class);
+                if (animComposer != null) {
+                    final Spatial mySpatial = finder.find(root, spatial);
+                    if (mySpatial != null) {
+                        //TODO: move attachments: have to scan through all
```

Comment:
Should we create an issue for that and maybe mention that in the TODO comment and vice versa?

Reply:
Yes, that is an old comment, but I think it's still valid. But it's probably a rare use case.